pkg/cmd/server/origin/controller/hpa.go

```diff
@@ -0,0 +1,43 @@
+package controller
+
+import (
+	hpacontroller "k8s.io/kubernetes/pkg/controller/podautoscaler"
+	hpametrics "k8s.io/kubernetes/pkg/controller/podautoscaler/metrics"
+
+	osclient "github.com/openshift/origin/pkg/client"
+	"github.com/openshift/origin/pkg/cmd/server/bootstrappolicy"
+)
+
+// NB: this is funky -- it's actually a Kubernetes controller, but we run it as an OpenShift controller in order
+// to get a handle on OpenShift clients, so that our delegating scales getter can work.
+
+type HorizontalPodAutoscalerControllerConfig struct {
+	HeapsterNamespace string
+}
+
+func (c *HorizontalPodAutoscalerControllerConfig) RunController(ctx ControllerContext) (bool, error) {
+	hpaClient := ctx.KubeControllerContext.ClientBuilder.ClientOrDie(bootstrappolicy.InfraHorizontalPodAutoscalerControllerServiceAccountName)
+	hpaOriginClient := ctx.ClientBuilder.DeprecatedOpenshiftClientOrDie(bootstrappolicy.InfraHorizontalPodAutoscalerControllerServiceAccountName)
```

Comment:
/me facepalms

Wow, that's the second brainfart I've had while working on this PR... :-/